cms/static/cms/js/modules/cms.structureboard.js

```diff
@@ -1295,16 +1298,108 @@ class StructureBoard {
         toolbar.prependTo(document.body);
         CMS.API.Toolbar._refreshMarkup(newToolbar);
 
-        $('.cms-structure-content').scrollTop(structureScrollTop);
+        this.addJsScriptsNeededForRender(new_scripts, old_scripts);
 
+        $('.cms-structure-content').scrollTop(structureScrollTop);
         Plugin._refreshPlugins();
-
-        Helpers._getWindow().dispatchEvent(new Event('load'));
         $(Helpers._getWindow()).trigger('cms-content-refresh');
 
         this._loadedContent = true;
     }
 
+    /**
+     * Checks if new scripts with the class 'cms-execute-js-to-render' exist
+     * and if the were present befor. If they weren't present before the will be downloaded
```

Comment:
and if the**y** were present befor**e**. If they weren't present before **-** the**y** will be downloaded